security/transport/transaction.py

```diff
@@ -80,3 +82,40 @@ def is_active_logged_requests(using=None):
     :return: True if block of code is surrounded with atomic_log operator
     """
     return hasattr(get_connection(using), 'output_logged_requests') and get_connection(using).output_logged_requests
+
+
+
+def get_all_request_related_objects():
+    """
+    Return all related selected with log_request_related_objects decorator.
+    :return: list of request related objects
+    """
+    connection = get_connection()
+    related_objects_list = getattr(connection, 'related_objects_list', [])
+    return list(itertools.chain(*related_objects_list))
+
+
+class LogRequestRelatedObjects(ContextDecorator):
+    """
+    Context decorator that adds related objects to all requests inside code block
+    """
+
+    def __init__(self, related_objects):
+        self.related_objects = related_objects
+
+    def __enter__(self):
+        connection = get_connection()
+        related_objects_list = getattr(connection, 'related_objects_list', [])
+        related_objects_list.append(self.related_objects)
+        connection.related_objects_list = related_objects_list
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        connection = get_connection()
+        connection.related_objects_list.pop()
```

Comment:
You are popping just the last object, in case multiple objects were passed, you are not removing them all. Maybe you could extend you test to verify this? I.e. make the request again without the decorator.

Reply:
you can use context processor inside context processor

Reply:
but yes I will add test